torch/linalg/__init__.py

```diff
@@ -184,3 +184,94 @@
     >>> torch.allclose(torch.tensordot(a, x, dims=x.ndim), b, atol=1e-6)
     True
 """)
+
+
+qr = _add_docstr(_linalg.linalg_qr, r"""
+qr(input, mode='reduced', *, out=None) -> (Tensor, Tensor)
+
+Computes the QR decomposition of a matrix or a batch of matrices :attr:`input`,
+and returns a namedtuple (Q, R) of tensors such that :math:`\text{input} = Q R`
+with :math:`Q` being an orthogonal matrix or batch of orthogonal matrices and
+:math:`R` being an upper triangular matrix or batch of upper triangular matrices.
+
+Depending on the value of :attr:`mode` this function returns the reduced or
+complete QR factorization. See below for a list of valid modes.
+
+.. warning:: **Differences with** :meth:`~torch.qr`:
+
+             * ``torch.qr`` takes a boolean parameter ``some`` instead of ``mode``:
+
+               - ``some=True`` is equivalent of ``mode='reduced'``: both are the
+                 default
+
+               - ``some=False`` is equivalent of ``mode='complete'``.
+
+             **Differences with** ``numpy.linalg.qr``:
+
+             * ``mode='raw'`` is not implemented
+
+             * when ``mode='r'`` numpy returns a single array `R`, whereas
+               pytorch returns a tuple `(Q, R)`, where `Q` is an empty
+               tensor.
+
+
+.. warning::
+          Backpropagation is not supported for ``mode='r'``. Use ``mode='reduced'`` instead.
+
+          If you plan to backpropagate through QR, note that the current backward implementation
+          is only well-defined when the first :math:`\min(input.size(-1), input.size(-2))`
+          columns of :attr:`input` are linearly independent.
+          This behavior will propably change once QR supports pivoting.
+
+.. note:: precision may be lost if the magnitudes of the elements of :attr:`input`
```

Comment:
I wonder if this note is precise enough to be helpful? Do you think it's worth keeping?

Reply:
honestly, I don't know. This was just a copy&paste from the docs of `torch.qr`

Reply:
Let's remove it for now.

Reply:
removed by both `qr` and `linalg.qr` in commit c18410e329